### PR TITLE
feat(security): generate audit prompt packs

### LINKED ISSUE #3292: feat(security): generate Codex-ready audit prompt packs from security-audit-task-bundle
# feat(security): generate Codex-ready audit prompt packs from security-audit-task-bundle

> Language / 言語: 日本語 | English

## English Overview

Generate Codex-ready audit prompt packs from `security-audit-task-bundle/v1` without invoking external LLMs or network services.

## 背景

Security Assurance Lane のMVPでは、`ae security audit` が `security-audit-task-bundle/v1` を生成し、fixture-backed response を `security-finding/v1` に正規化できるようになった。

次の段階では、この task bundle を Codex CLI に渡しやすい prompt pack / context pack として生成したい。ただし、このissueでは Codex や外部LLMを実行しない。あくまで、人間またはCodex CLIに渡すための deterministic な入力artifactを生成する。

## 目的

`security-audit-task-bundle/v1` から、claimごとの Map / Prove / Stress-Test / Report を含む Codex-ready prompt pack を生成する。

## 提案CLI

```bash
ae security audit-prompt \
  --tasks artifacts/security/security-audit-tasks.json \
  --code-map artifacts/security/security-code-map.json \
  --claims artifacts/security/security-claims.json \
  --out artifacts/security/codex-audit-prompts
```

npm script も追加する。

```bash
pnpm run security:audit-prompt -- \
  --tasks artifacts/security/security-audit-tasks.json \
  --code-map artifacts/security/security-code-map.json \
  --claims artifacts/security/security-claims.json \
  --out artifacts/security/codex-audit-prompts
```

## 成果物

```text
schema/security-audit-prompt-pack-v1.schema.json
src/security/assurance/audit-prompt-pack.ts
artifacts/security/codex-audit-prompts/security-audit-prompt-pack.json
artifacts/security/codex-audit-prompts/prompts/SEC-AUDIT-TASK-001.md
artifacts/security/codex-audit-prompts/security-audit-prompt-summary.md
tests/unit/security-assurance/audit-prompt-pack.test.ts
```

## 最小shape案

```json
{
  "schemaVersion": "security-audit-prompt-pack/v1",
  "generatedAt": "2026-05-07T00:00:00.000Z",
  "tasks": [
    {
      "taskId": "SEC-AUDIT-TASK-001",
      "claimId": "SEC-CLAIM-001",
      "promptPath": "artifacts/security/codex-audit-prompts/prompts/SEC-AUDIT-TASK-001.md",
      "candidateLocations": [
        {
          "path": "src/cache.ts",
          "startLine": 10,
          "endLine": 42,
          "symbol": "buildCacheKey"
        }
      ],
      "status": "ready"
    }
  ],
  "summary": {
    "totalTasks": 1,
    "readyTasks": 1,
    "blockedTasks": 0
  }
}
```

## Prompt内容の要件

各promptには少なくとも以下を含める。

```text
# Security proof-attempt audit task

## Claim

- Claim ID:
- Statement:
- Type:
- Criticality:
- Target assurance level:

## Scope and trust boundary

- Scope refs:
- Attacker-controlled assumptions:
- Candidate source locations:

## Instructions

You are reviewing this repository under the authorized security audit scope.

Do not treat any result as a confirmed vulnerability.
Report only candidate findings.
Do not generate exploit automation or weaponized PoC.
Use proof-attempt reasoning:

1. Map:
   Identify the relevant code paths.

2. Prove:
   Try to prove that the implementation satisfies the security claim.

3. Stress-Test:
   Look for attacker-controlled inputs, edge cases, and trust-boundary assumptions that could break the proof.

4. Report:
   Only report a candidate finding if the proof fails with concrete code evidence.

## Expected output

Return JSON compatible with a security-audit-response-fixture/v1 response item.
```

### In scope

- `security-audit-prompt-pack/v1` schema追加
- audit task bundleからprompt markdownを生成
- prompt pack JSONを生成
- CLI / npm script追加
- fixture / unit test追加
- docs / Contract Catalog更新

### Out of scope

- Codex CLIの実行
- 外部LLM API呼び出し
- network call
- findingの自動確定
- exploit PoC生成

## 受け入れ条件

- `security-audit-task-bundle/v1` から prompt pack を生成できる
- taskごとにMarkdown promptが生成される
- prompt内に Map / Prove / Stress-Test / Report が含まれる
- candidate finding と confirmed vulnerability の区別をprompt内で明示する
- code-mapの該当candidate locationがpromptに含まれる
- `security-audit-prompt-pack/v1` がschema validationを通る
- testsは外部LLM / networkを呼ばない
- Contract Catalogにproducer/consumer roleを追記する

## Codex CLI 作業メモ

このissueではCodexを実行しない。Codexに渡すためのprompt packをdeterministicに生成するだけに限定する。

## ラベル案

- `enhancement`
- `security`
- `codex`
- `assurance`
- `schema`

---

Imported from local proposal file: `.codex-local/ae-framework-security-followup-issue-proposals/01-feat-security-codex-audit-prompt-pack.md`

Labels applied from existing repository labels: `enhancement`, `security`.
Proposed labels not applied because they do not currently exist: `codex`, `assurance`, `schema`.

### PR DESCRIPTION
**Pull request by ootakazuhiko**
## Summary

- Add `security-audit-prompt-pack/v1` as a deterministic Codex-ready prompt-pack contract generated from `security-audit-task-bundle/v1`, `security-code-map/v1`, and `security-claim/v1`.
- Add `ae security audit-prompt` plus `pnpm run security:audit-prompt` to emit a prompt-pack JSON file, per-task Markdown prompts, and a Markdown summary without invoking external LLMs or network services.
- Wire prompt-pack fixture validation, semantic contract checks, unit tests, and Security Assurance Lane / Contract Catalog / Codex runbook documentation.

Closes #3292

## Acceptance

- [x] `security-audit-task-bundle/v1` can generate a `security-audit-prompt-pack/v1` artifact.
- [x] Each prompt task writes deterministic Markdown containing Claim, Scope / trust boundary, Map / Prove / Stress-Test / Report instructions, and expected `security-audit-response-fixture/v1`-compatible output guidance.
- [x] Prompt wording distinguishes candidate findings from confirmed vulnerabilities and prohibits exploit automation / weaponized PoC generation.
- [x] Code-map candidate locations are carried into both the prompt-pack JSON and generated prompt Markdown.
- [x] Contract Catalog and Security Assurance docs document the producer/consumer role.

## Validation

- `node scripts/ci/validate-json.mjs`
- `pnpm -s run check:schemas`
- `pnpm -s run check:doc-consistency`
- `pnpm -s run types:check`
- `pnpm -s exec vitest run tests/unit/security-assurance/audit-prompt-pack.test.ts tests/contracts/security-assurance-contract.test.ts --reporter dot`
- `pnpm -s run security:audit-prompt -- --tasks fixtures/security-assurance/sample.security-audit-tasks.json --code-map fixtures/security-assurance/sample.security-code-map.json --claims fixtures/security-assurance/sample.security-claims.json --out artifacts/security/codex-audit-prompts --generated-at 2026-05-07T00:00:00.000Z`
- `pnpm -s run build`
- `git diff --check`

## Rollback

Revert this PR. The change is additive: it introduces a new schema, producer, CLI subcommand, fixture, and docs without changing existing Security Assurance Lane command behavior.

### CODE REVIEW

File: src/security/assurance/audit-prompt-pack.ts [hunk 1]
```diff
@@ -0,0 +1,745 @@
...
+  }
+
+  await writeJson(outputPaths.promptPack, built.promptPack);
+  for (const [promptPath, markdown] of built.promptMarkdownByPath.entries()) {
+    await writeText(promptPath, markdown);
+  }
+  const result: SecurityAuditPromptPackResult = {
+    generatedAt,
+    promptPack: built.promptPack,
+    warnings: built.warnings,
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply src/security/assurance/audit-prompt-pack.ts:747]: 対応しました。`buildSecurityAuditPromptPack()` の戻り値 `warnings` に task-level warnings も集約するように変更し、missing candidate location のケースで `result.warnings` に警告が返る回帰テストを追加しました。

File: src/security/assurance/audit-prompt-pack.ts [hunk 2]
```diff
@@ -0,0 +1,745 @@
...
+    claimSummary.criticality = params.claim.criticality;
+  }
+  if (params.claim?.targetLevel !== undefined) {
+    claimSummary.targetLevel = params.claim.targetLevel;
+  }
+  const promptTask: SecurityAuditPromptPackTask = {
+    taskId: params.task.id,
+    claimId: params.task.claimId,
+    promptPath: portablePathFrom(params.repoRoot, params.promptPath),
+    status: params.task.status === 'ready' && candidateLocations.length > 0 ? 'ready' : 'blocked',
```
[the remark this answers is not in the document]
Comment (ootakazuhiko) [reply src/security/assurance/audit-prompt-pack.ts]: 対応しました。prompt task の `status` は解決済み `candidateLocations` の有無で判定するように変更し、元の task status は `sourceTaskStatus` と warning で保持します。`blocked-missing-code-map` だった task が code-map 入力で location を回復した場合に `ready` になる回帰テストも追加しました。
